packages/cloud_firestore/cloud_firestore/lib/src/firestore.dart

```diff
@@ -98,4 +101,14 @@ class Firestore {
           host: host,
           sslEnabled: sslEnabled,
           cacheSizeBytes: cacheSizeBytes);
+
+  @override
+  int get hashCode => _delegate.app.name.hashCode;
```

Comment:
```suggestion
  int get hashCode => _delegate.app.hashCode;
```

@slightfoot 

The `hashCode` of `FirebaseApp` is implemented appropriately, so `.name` seems redundant.

- https://github.com/FirebaseExtended/flutterfire/blob/5e380cf85fde7eefa91d520bdb3ae0bfeb2d1282/packages/firebase_core/firebase_core/lib/src/firebase_app.dart#L86